### ISSUE TITLE
[java] [regression from 6.55] UnusedLocalVariable false-positive in try with resource

### ISSUE THREAD

**Issue by candrews** [pmd/pmd#4887]
**Affects PMD Version:** 

7.0.0

**Rule:**

UnusedLocalVariable

https://docs.pmd-code.org/pmd-doc-7.0.0/pmd_rules_java_bestpractices.html#unusedlocalvariable

**Description:**

When using try-with-resources where the Closeable isn't used, PMD falsely reports the variable as being unused. 

**Code Sample demonstrating the issue:**

```java
try (var providerCloseable = MDC.putCloseable("something", "a");
		var kindCloseable = MDC.putCloseable("other", "b");
		var actionCloseable = MDC.putCloseable("more", "c");
		var categoryCloseable = MDC.putCloseable("things", "d");
		var typeCloseable = MDC.putCloseable("even", "e");
		var outcomeCloseable = MDC.putCloseable("odd", "f");
		) {
	log.info("Log");
}
```

**Expected outcome:**

PMD reports violations at lines 1-6, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Gradle.

**Comment by oowekyala** [comment]
This is working as intended. This behavior is intended to have you mark these variables clearly as unused by design. You can do that with a `@SuppressWarnings` annotation, or rename these variables with a name that starts with `unused` or `ignored`. In Java 21 (preview) and 22 you can also name them all `_`.
